server/validate.go

```diff
@@ -70,9 +94,41 @@ func validateGetEntryAndProofRequest(req *trillian.GetEntryAndProofRequest) erro
 	return nil
 }
 
+func validateQueueLeafRequest(req *trillian.QueueLeafRequest) error {
+	if req.Leaf == nil {
+		return status.Error(codes.InvalidArgument, "QueueLeafRequest.Leaf empty")
+	}
+	if err := validateLogLeaf(req.Leaf); err != nil {
+		return status.Errorf(codes.InvalidArgument, "QueueLeafRequest.Leaf.%v", err)
```

Comment:
Curious about the `.%v` for `err` - why not the more usual `...: %v` style?

Reply:
validateLogLeaf returns a message that chains nicely with this format. Added comments / docs to make it clearer.